File: benchmarks/bench_abs.zig
const std = @import("std");

fn absi(x: i32) i32 {
    return if (x < 0) -x else x;
}

pub fn main() u8 {
    var r: i32 = 0;
    var i: i32 = -500000000;
    while (i < 500000000) : (i += 1) {
        r = absi(i);
    }
    return @intCast(r & 0xFF);
}

File: benchmarks/bench_abs.c
#include <stdio.h>
#include <stdint.h>

static int32_t absi(int32_t x) { return x < 0 ? -x : x; }

int main(void) {
    volatile int32_t r = 0;
    for (int i = -500000000; i < 500000000; i++) {
        r = absi(i);
    }
    return (int)(r & 0xFF);
}
